src/components/SiteVersion.js

```diff
@@ -1,26 +1,76 @@
-import React from 'react';
-import { Link } from '@makerdao/ui-components-core';
+import React, { useState, useEffect } from 'react';
+import { Grid, Box, Link, Text } from '@makerdao/ui-components-core';
+import 'whatwg-fetch';
 import styled from 'styled-components';
-import { message } from '../static/version.json';
 
-const VersionLink = styled(Link)`
+const VersionText = styled(Text)`
   font-size: 0.75em;
   color: lightgray;
+`;
+const VersionLink = styled(Link)`
   &:hover {
-    color: gray;
+    ${VersionText} {
+      color: gray;
+    }
+    ${Grid} {
+      border: 1px solid gray;
+      border-radius: 3px;
+    }
   }
 `;
 
 const SiteVersion = () => {
-  const commitId = message.substr(0, 7);
+  const {
+    NOW_GITHUB_REPO,
+    NOW_GITHUB_ORG,
+    NOW_GITHUB_COMMIT_SHA,
+    NOW_GITHUB_COMMIT_AUTHOR_NAME,
+    NOW_GITHUB_COMMIT_REF
+  } = require('../static/version.json');
+
+  const [commitMsg, setCommitMsg] = useState('');
+  const [commitDate, setCommitDate] = useState('');
+  const apiUrl = `https://api.github.com/repos/${NOW_GITHUB_ORG}/${NOW_GITHUB_REPO}/git/commits/${NOW_GITHUB_COMMIT_SHA}`;
+
+  useEffect(() => {
+    const fetchCommitMsg = async () => {
+      const rawJson = await fetch(apiUrl);
+      const commitObj = await rawJson.json();
+      const {
+        message,
+        committer: { date }
+      } = commitObj;
+      const d = new Date(date);
+      setCommitDate(d.toLocaleString());
+      setCommitMsg(message);
+    };
+    fetchCommitMsg();
+  }, [apiUrl]);
 
+  const commitUrl = `https://github.com/${NOW_GITHUB_ORG}/${NOW_GITHUB_REPO}/commit/${NOW_GITHUB_COMMIT_SHA}`;
   return (
-    <VersionLink
-      href={`https://github.com/makerdao/mcd-cdp-portal/tree/${commitId}`}
-      target="_blank"
-    >
-      {`Site version: ${message}`}
-    </VersionLink>
+    <Box>
+      {commitMsg !== '' ? (
+        <VersionLink href={commitUrl} target="_blank">
+          <Grid p="s" gridTemplateColumns="1fr 2fr">
+            <VersionText>Author:</VersionText>
+            <VersionText>{NOW_GITHUB_COMMIT_AUTHOR_NAME}</VersionText>
+
+            <VersionText>Commit:</VersionText>
+            <VersionText>{commitMsg}</VersionText>
+
+            <VersionText>Hash:</VersionText>
+            <VersionText>{NOW_GITHUB_COMMIT_SHA.substring(0, 10)}</VersionText>
+
+            <VersionText>Branch:</VersionText>
+            <VersionText>{NOW_GITHUB_COMMIT_REF}</VersionText>
+
+            <VersionText>Uploaded:</VersionText>
+            <VersionText>{commitDate}</VersionText>
+          </Grid>
+        </VersionLink>
+      ) : null}
+    </Box>
```

Comment:
Actually, one small thing to mention @sirromdev : this renders behind the feedback button, so you might want to add some bottom padding to the component to make it readable.

<img width="299" alt="Screen Shot 2019-06-07 at 2 15 10 PM" src="https://user-images.githubusercontent.com/20496459/59131173-d9f93680-892e-11e9-8622-7a6b58ae6746.png">

Reply:
:+1:

Reply:
Is the feedback button not an overlay? So depending on the dimensions of your screen it may and may not cover the box?